target/scripts/startup/setup-stack.sh

```diff
@@ -1,7 +1,21 @@
 #!/bin/bash
 
+declare -a FUNCS_SETUP
+
+function _register_setup_function
+{
+  FUNCS_SETUP+=("${1}")
+  _log 'trace' "${1}() registered"
+}
+
 function _setup
 {
+  while read -r FILE
+  do
+    # shellcheck source=/dev/null
+    source "${FILE}"
+  done < <(find /usr/local/bin/setup.d/ -type f)
```

Comment:
Generally bad practice - files including spaces for example will not be handled correctly. You should add `-print0` to the find command and define a delimiter `-d $'\0'` in the `read` command or just use globbing instead:

`for i in /usr/local/bin/setup.d/*; do source $i; done`

Reply:
See #3124.

Reply:
You are right, I overlooked that there is also a subdirectory and not just files.